pkgs/top-level/all-packages.nix

```diff
@@ -12137,6 +12138,7 @@ with pkgs;
       num =
         if (with stdenv.targetPlatform; isVc4 || libc == "relibc") then 6
         else if (stdenv.targetPlatform.isAarch64 && stdenv.isDarwin) then 11
+        else if (stdenv.targetPlatform.isx86_64 && stdenv.isLinux) then 11
```

Comment:
Why adding yet another exception instead of changing the default 10 below?

Reply:
I have no access to x86_64-darwin nor aarch64-linux platforms.

PR was built and tested locally on x86_64-linux before proposing it alongsides binutils 2.38 bump that also got into staging. (Which had issues on x86_64-darwin while x86_64-linux was working fine at the time; I though that binutils bump was also only going to happen for x86_64-linux)

Reply:
But the default does not apply to those platforms.

Reply:
Well, darwin doesn't use gcc as the default compiler, but maybe for `x86_64-darwin` there is some difference here.

Reply:
The only GCC version not pinned would be x86_64-darwin. If a @NixOS/darwin-maintainers has bandwidth to bump GCC on that platform then that route seems better but don't want to push on a platform I neither use nor can help with.

Reply:
I had i686-linux in mind mainly, and the overall complexity of the switch.  Anyway, thanks for explanation.  I'm looking into this.

Reply:
Hmm, I'll probably leave it this way now and maybe come back once it reaches master.

Reply:
I've tested `x86_64-darwin` and it seems ok: https://github.com/NixOS/nixpkgs/pull/168093. Unfortunately it times out on ofborg.

Reply:
OK, `wine` wouldn't build if I switched i686 to gcc11 as well, with errors during `mingw-w64-i686-w64-mingw32-9.0.0.drv`, and aarch64-linux WIP is separate in #167726.  So I guess the rest of the mess will remain for now.